### DISCUSSION TITLE
how to configure dex as https protocol

### DISCUSSION THREAD

**Discussion by gy95** [dexidp/dex#2013] [Q&A] [answered]
my config-dev.yaml is as follows, I comment for HTTP options and uncomment for HTTPS options , but it seems not to work in HTTPS mode.  When a HTTPS client request, dex gives a HTTP response.

  web:
    #http: 127.0.0.1:5556
    # Uncomment for HTTPS options.
    https: 0.0.0.0:5554
    tlsCert: /home/dex/ssl/ssl.crt
    tlsKey: /home/dex/ssl/ssl.key

how to fix this problem?

after starting server, i run this command as follows
./bin/example-app --listen https://127.0.0.1:5554 --issuer https://127.0.0.1:5554/dex --issuer-root-ca /home/dex/ssl/127.0.0.1.crt
error is :
failed to query provider "https://127.0.0.1:5554/dex": Get "https://127.0.0.1:5554/dex/.well-known/openid-configuration": http: server gave HTTP response to HTTPS client

**Comment by nabokihms** [comment, marked as answer]
Hello! Thanks for reporting this. 

Are you trying to run dex from the master branch? The HTTPS server seems to be broken in the master right now. This is a regression, and I am working on a fix. The workaround is to use one of the tag releases, e.g., v2.27.0.

**Reply by gy95** [reply]
yes, I run dex from the master branch.  And v2.27.0 it works well

**Reply by nabokihms** [reply]
PR with the fix
https://github.com/dexidp/dex/pull/2018
